third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -254,8 +254,34 @@ function messageCallback(message) {
 
 // Methods for interacting with the record/replay driver.
 
+// Track all current execution contexts so that any scripts that we
+// inject via evaluatePrivilegd can know what contexts are available.
+const gExecutionContexts = new Map();
+const gContextChangeCallbacks = new Set();
+
 initMessages();
 addEventListener("Runtime.consoleAPICalled", onConsoleAPICall);
+addEventListener("Runtime.executionContextCreated", ({ context }) => {
+  gExecutionContexts.set(context.id, context);
+  for (const callback of gContextChangeCallbacks) {
+    callback(context, "add");
```

Comment:
Can we wrap all the callback invokes in a try/catch?  It seems like we'd have some other interface for adding things to `gContextChangeCallbacks`, and that code could possibly throw.  Not sure what happens to exceptions at the top level of that script.

Reply:
I guess the question there would be what happens if this throws, I guess I was assuming it'd just crash the whole process

Reply:
But yeah happy to add that.

>  It seems like we'd have some other interface for adding things to gContextChangeCallbacks

Was thinking things would just access the Set directly but happy either way. The main thing with `evaluatePrivileged` is that we haven't at all defined what it should and shouldn't be touching.